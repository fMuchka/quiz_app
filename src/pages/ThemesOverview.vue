<template>
    <div id="overviewWrapper">
        <div v-for="(item, index) in themes" 
            :key="index"
        >
        <button  
            class="themeButton"
            @click="openFirstQuestion(index)"
            :style="completedThemeStyle(index)"
        >
            {{ item.text }}  
        </button>

        <button
            class="answersButton"
            @click="openThemeAnswers(index)"
            :disabled="!isThemeCompleted(index)"
        >
            OdpovÄ›di
        </button>
        </div>
        
    </div>
</template>

<script>
export default {
    data(){
        return{
            themes: this.$store.getters.quiz.themes
        }
    },

    methods: {
        openFirstQuestion(index){
            // go to first question of selected theme
            this.$store.commit("setCurrentTheme", index);
            this.$store.commit("firstQuestion");
            
            this.$router.push({   
                        name: 'questionslide'  
            });
        },

        openThemeAnswers(index){
            this.$store.commit("setCurrentTheme", index);
            
            this.$router.push({   
                        name: 'themeanswers'  
            });
        },

        completedThemeStyle(index){
            const completedThemes = this.$store.state.completedThemes;
            
            if (completedThemes[index] === true) {
               return "background-color: green;"; 
            }

            return "";
        },

        isThemeCompleted(index){
            const completedThemes = this.$store.state.completedThemes;
            
            if (completedThemes[index] === true) {
               return true; 
            }

            return false;
        }
    }
}
</script>

<style>
#overviewWrapper {
    display: flex;
    flex-flow: wrap;
    height: 100%;
    width: 80%;
    margin: auto;
    flex-direction: column;
    overflow-y: auto;
}

.themeButton {
    background-color: var(--secondary-color);
    color: white;
    border: none;
    width: 40%;
    height: 160px;
    font-size: 64px;
    margin: auto;
}

.themeButton:hover {
    cursor: pointer;
    color: 003049;
}

#overviewWrapper div {
    display: flex;
    flex-direction: column;
    margin: 50px 0;
}

.answersButton {
    width: 40%;
    display: flex;
    margin: auto;
    justify-content: center;
    height: 50px;
    align-items: center;
    cursor: pointer;
}
</style>