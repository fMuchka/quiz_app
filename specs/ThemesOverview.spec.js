import Vuex from "vuex";
import { mount, createLocalVue } from "@vue/test-utils";

import ThemesOverview from "../src/pages/ThemesOverview.vue"
import mainStore from "../src/store.js";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("ThemesOverview Page:", () => {
    let store;

    beforeEach(() => {
        mainStore.state.quiz = { theme: [{ title: "1" }, { title: "2" }, { title: "3" }] };
        store = new Vuex.Store(mainStore);
    });

    it("matches snapshot", () => {
        const wrapper = mount(ThemesOverview, {
            localVue,
            store
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it("creates Themes based on Store.state.quiz", () => {
        const wrapper = mount(ThemesOverview, {
            localVue,
            store
        });

        const testThemes = store.state.quiz.theme;
        const pageElements = wrapper.vnode.children;

        expect(pageElements.length).toBe(testThemes.length);

        for (let i = 0; i < pageElements.length; i++){
            expect(pageElements[i].children[0].text.includes(testThemes[i].title)).toBeTruthy()
        }
    });

    it('opens first question on click', () => {
        const wrapper = mount(ThemesOverview, {
            localVue,
            store
        });
        
        const openSpy = jest.spyOn(wrapper.vm, "openFirstQuestion");
        const button = wrapper.find("button");
        button.trigger('click');

        expect(openSpy).toHaveBeenCalled();

        expect(store.state.currentIndexes.question).toBe(0);
    });
})