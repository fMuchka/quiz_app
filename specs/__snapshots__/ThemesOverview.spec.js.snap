// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemesOverview Page: matches snapshot 1`] = `
<div
  id="overviewWrapper"
>
  <button>
    
        1  
    
  </button>
  <button>
    
        2  
    
  </button>
  <button>
    
        3  
    
  </button>
</div>
`;
